waveorder/io/reader.py

```diff
@@ -0,0 +1,219 @@
+import numpy as np
+import os
+import zarr
+from tifffile import TiffFile
+from copy import copy
+import logging
+
+
+# replicate from aicsimageio logging mechanism
+###############################################################################
+
+# modify the logging.ERROR level lower for more info
+# CRITICAL
+# ERROR
+# WARNING
+# INFO
+# DEBUG
+# NOTSET
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="[%(levelname)4s: %(module)s:%(lineno)4s %(asctime)s] %(message)s",
+)
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class MicromanagerReader:
+
+    def __init__(self, folder: str, extract_data: bool = False, log_level: int = logging.ERROR):
+        """
+        reads ome-tiff files into zarr or numpy arrays
+        Strategy:
+            1. search the file's omexml metadata for the "master file" location
+            2. load the master file
+            3. read micro-manager metadata into class attributes
+
+        :param folder: str
+            folder or file containing all ome-tiff files
+        :param extract_data: bool
+            True if ome_series should be extracted immediately
+        :param log_level: int
+            One of 0, 10, 20, 30, 40, 50 for NOTSET, DEBUG, INFO, WARNING, ERROR, CRITICAL respectively
+
+        """
+
+        logging.basicConfig(
+            level=log_level,
+            format="[%(levelname)4s: %(module)s:%(lineno)4s %(asctime)s] %(message)s",
+        )
+        self.log = logging.getLogger(__name__)
+
+        self._positions = {}
+        self.mm_meta = None
+        self.stage_positions = 0
+        self.height = 0
+        self.width = 0
+        self.frames = 0
+        self.slices = 0
+        self.channels = 0
+        self.shape = (self.frames, self.slices, self.channels, self.height, self.width)
```

Comment:
self.shape is not updated after creating reader object

Reply:
good catch, fixed